shell/platform/darwin/macos/framework/Source/FlutterChannelKeyResponderTest.mm

```diff
@@ -35,258 +69,216 @@
+  __block NSMutableArray<NSNumber*>* responses = [[NSMutableArray<NSNumber*> alloc] init];
+  FakeMessageChannel* channel = [[FakeMessageChannel alloc] init];
   FlutterChannelKeyResponder* responder =
-      [[FlutterChannelKeyResponder alloc] initWithChannel:mockKeyEventChannel];
+      [[FlutterChannelKeyResponder alloc] initWithChannel:channel];
 
-  // Initial empty modifiers. This can happen when user opens window while modifier key is pressed
-  // and then releases the modifier. No events should be sent, but the callback
-  // should still be called.
+  // Initial empty modifiers.
+  //
+  // This can happen when user opens window while modifier key is pressed and then releases the
+  // modifier. No events should be sent, but the callback should still be called.
   // Regression test for https://github.com/flutter/flutter/issues/87339.
-  [responder handleEvent:keyEvent(NSEventTypeFlagsChanged, 0x100, @"", @"", FALSE, 60)
+  channel.nextResponse = @{@"handled" : @YES};
+  [responder handleEvent:CreateKeyEvent(NSEventTypeFlagsChanged, 0x100, @"", @"", NO, 60)
                 callback:^(BOOL handled) {
                   [responses addObject:@(handled)];
                 }];
 
-  EXPECT_EQ([messages count], 0u);
-  EXPECT_EQ([responses count], 1u);
-  EXPECT_EQ([responses[0] boolValue], TRUE);
+  EXPECT_EQ([channel.messages count], 0u);
+  ASSERT_EQ([responses count], 1u);
+  EXPECT_EQ([responses[0] boolValue], YES);
   [responses removeAllObjects];
 
   // Key down
-  [responder handleEvent:keyEvent(NSEventTypeKeyDown, 0x100, @"a", @"a", FALSE, 0)
+  channel.nextResponse = @{@"handled" : @YES};
+  [responder handleEvent:CreateKeyEvent(NSEventTypeKeyDown, 0x100, @"a", @"a", NO, 0)
                 callback:^(BOOL handled) {
                   [responses addObject:@(handled)];
                 }];
 
-  EXPECT_EQ([messages count], 1u);
-  EXPECT_STREQ([[messages lastObject][@"keymap"] UTF8String], "macos");
-  EXPECT_STREQ([[messages lastObject][@"type"] UTF8String], "keydown");
-  EXPECT_EQ([[messages lastObject][@"keyCode"] intValue], 0);
-  EXPECT_EQ([[messages lastObject][@"modifiers"] intValue], 0x0);
-  EXPECT_STREQ([[messages lastObject][@"characters"] UTF8String], "a");
-  EXPECT_STREQ([[messages lastObject][@"charactersIgnoringModifiers"] UTF8String], "a");
+  ASSERT_EQ([channel.messages count], 1u);
+  EXPECT_STREQ([[channel.messages lastObject][@"keymap"] UTF8String], "macos");
+  EXPECT_STREQ([[channel.messages lastObject][@"type"] UTF8String], "keydown");
+  EXPECT_EQ([[channel.messages lastObject][@"keyCode"] intValue], 0);
+  EXPECT_EQ([[channel.messages lastObject][@"modifiers"] intValue], 0x0);
+  EXPECT_STREQ([[channel.messages lastObject][@"characters"] UTF8String], "a");
+  EXPECT_STREQ([[channel.messages lastObject][@"charactersIgnoringModifiers"] UTF8String], "a");
 
-  EXPECT_EQ([responses count], 1u);
-  EXPECT_EQ([[responses lastObject] boolValue], TRUE);
+  ASSERT_EQ([responses count], 1u);
```

Comment:
If I understand you correctly, these should be changed because we rely on it later in the test, so it will segfault otherwise?

Reply:
Correct -- this one was changed to ASSERT, which is what we want.